sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -132,7 +132,7 @@ namespace Azure { namespace Core { namespace Http {
      * @brief When `true`, allows an invalid certificate authority. If this flag is set, the
      * application does not receive a WINHTTP_CALLBACK_STATUS_FLAG_INVALID_CA callback.
      */
-    bool IgnoreUnknownServerCert = false;
```

Comment:
As @ahsonkhan pointed out, this boolean doesn't actually enable ignoring unknown server certificates. Instead this option configures the WinHTTP transport to ignore invalid certificate authorities.

Also, I'm trying to understand what an "Unknown Certificate" means. An X.509 certificate is identified by two fields: Issuer and Subject - the subject identifies the specific certificate, the Issuer identifies which CA issued the certificate.

Is an "Unknown Certificate" a certificate where the CA isn't known (or the root CA isn't one of the known root CAs)? Or is it a certificate where the subject doesn't match the server's IP address/DNS address? Or is it something else?

To me, the comment for this boolean is pretty clear: This option ignores invalid certificate authorities. But it says nothing about the validity of the leaf certificate being used to secure the connection.

Can the name of this field be updated to reflect the comment? Something like @ahsomething's suggestion of "IgnoreInvalidCertificateAuthorities"?

Reply:
Here's where the option comes and maps to:
![image](https://user-images.githubusercontent.com/24213737/160895696-d2f8b9e7-1658-4d2a-9970-ba57736eabd2.png)

https://docs.microsoft.com/en-us/windows/win32/wininet/option-flags

`Ignores unknown certificate authority problems.`

Reply:
So why doesn't the field name match the option that it configures? Another way of asking the same question is: "Why is this option known as "IgnoreUnknownServerCertificate" instead of "IgnoreInvalidCertificateAuthorities"

Reply:
Just as a reference, here's a pointer to the previous conversation:
https://github.com/Azure/azure-sdk-for-cpp/pull/3434#discussion_r838082790

Reply:
> So why doesn't the field name match the option that it configures?

I think about multiple answers to that question, but none of them is for actually defend the current name v/s any other name :)

The simplest answer might be that there's this issue: https://github.com/Azure/azure-sdk-for-cpp/issues/3486 asking to update the name and it is mentioning the expected name. This PR is following that request.

Reply:
I don't have any preference. I am following what it requested from https://github.com/Azure/azure-sdk-for-cpp/issues/3486.  The ticket was assigned to me, so I am trying to fix it.

Please either update the ticket based on the name you guys would like to use, or directly push a commit to this PR, approve and I can merge.

Keep in mind azure-core release is due to 3/31, and this PR blocks the release.  So, please prioritize @RickWinter

Reply:
I don't have a strong preference on the names, both are acceptable.  If I had to choose I'd go with `IgnoreUnknownCertificateAuthority`.  Staying close to the WinHttp name makes a lot of sense given this is a WinHttp specific option.